src/lib.rs

```diff
@@ -207,395 +432,112 @@ impl Mmap {
-    pub fn ptr(&self) -> *const u8 {
-        self.inner.ptr()
-    }
-
-    /// Returns a pointer to the mapped memory.
-    ///
-    /// See `Mmap::as_mut_slice` for invariants that must hold when
-    /// dereferencing the pointer.
-    pub fn mut_ptr(&mut self) -> *mut u8 {
-        self.inner.mut_ptr()
-    }
-
-    /// Returns the memory mapped file as an immutable slice.
-    ///
-    /// ## Unsafety
-    ///
-    /// The caller must ensure that the file is not concurrently modified.
-    pub unsafe fn as_slice(&self) -> &[u8] {
-        slice::from_raw_parts(self.ptr(), self.len())
-    }
-
-    /// Returns the memory mapped file as a mutable slice.
-    ///
-    /// ## Unsafety
-    ///
-    /// The caller must ensure that the file is not concurrently accessed.
-    pub unsafe fn as_mut_slice(&mut self) -> &mut [u8] {
-        slice::from_raw_parts_mut(self.mut_ptr(), self.len())
-    }
-
-    /// Creates a splittable mmap view from the mmap.
-    pub fn into_view(self) -> MmapView {
-        let len = self.len();
-        MmapView { inner: Rc::new(UnsafeCell::new(self)),
-                   offset: 0,
-                   len: len }
-    }
-
-    /// Creates a thread-safe splittable mmap view from the mmap.
-    pub fn into_view_sync(self) -> MmapViewSync {
-        let len = self.len();
-        MmapViewSync { inner: Arc::new(UnsafeCell::new(self)),
-                       offset: 0,
-                       len: len }
-    }
-}
-
-impl fmt::Debug for Mmap {
-    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        write!(f, "Mmap {{ ptr: {:?}, len: {} }}", self.ptr(), self.len())
-    }
-}
-
-/// A view of a memory map.
-///
-/// The view may be split into disjoint ranges, each of which will share the
-/// underlying memory map.
-pub struct MmapView {
-    inner: Rc<UnsafeCell<Mmap>>,
-    offset: usize,
-    len: usize,
-}
-
-impl MmapView {
-
-    /// Split the view into disjoint pieces at the specified offset.
-    ///
-    /// The provided offset must be less than the view's length.
-    pub fn split_at(self, offset: usize) -> Result<(MmapView, MmapView)> {
-        if self.len < offset {
-            return Err(Error::new(ErrorKind::InvalidInput,
-                                  "mmap view split offset must be less than the view length"));
+    /// This method only changes the protection of the underlying mapping,
+    /// but it doesn't make an `Mmap` from an `MmapMut`, use
+    /// [`make_read_only()`](#method.make_read_only) method for that.
+    ///
+    /// # Errors
+    ///
+    /// This method returns `Err` when the underlying system call fails, which can happen for
+    /// a variety of reasons, such as when you don't have the necessary permissions for the file.
+    ///
+    /// This method *also* returns `Err` with `ErrorKind` set to `InvalidInput` if the specified
+    /// protection does not allow the mapping to be mutable.
+    pub fn set_protection(&mut self, protection: Protection) -> Result<()> {
```

Comment:
So this is just for switching from ReadWrite to ReadCopy and vice versa?  That seems kinda strange - we should probably document what that even means.  I'm surprised the OS allows it.